### PR TITLE
[FLINK-5763][state backends] (follow-up) Make savepoints self-contained and relocatable

### PR DESCRIPTION
**Pull request by StephanEwen**
## What is the purpose of the change

This PR is a set of follow-up suggestion to PR #11722 (Make savepoints self-contained and relocatable)

Please refer to the original PR for description and discussion.

### CODE REVIEW

File: flink-core/src/main/java/org/apache/flink/core/fs/EntropyInjector.java [hunk 1]
```diff
@@ -89,6 +89,10 @@ public static Path removeEntropyMarkerIfPresent(FileSystem fs, Path path) {
 
 	// ------------------------------------------------------------------------
 
+	public static boolean isEntropyInjecting(FileSystem fs) {
+		return getEntropyFs(fs) == null;
```
Comment (klion26) [inline flink-core/src/main/java/org/apache/flink/core/fs/EntropyInjector.java:93]: `getEntropyFs(fs) != null;`

Comment (StephanEwen) [reply flink-core/src/main/java/org/apache/flink/core/fs/EntropyInjector.java:93]: You are right, definitely. This hints that I need to add a test.

**Comment by StephanEwen** [comment]
Thanks for reviewing this. I will add a test for the entropy injection bug and then merge both PRs together.

**Comment by StephanEwen** [comment]
Manually merged in 4ee41bebc504880c7224ec29ffdd4cf28f5fc218

**Pull request closed without merging**